timemachine/fe/stored_arrays.py

```diff
@@ -0,0 +1,140 @@
+import io
+import shutil
+import tempfile
+import weakref
+from itertools import count
+from pathlib import Path
+from typing import Iterator, List, NoReturn, Sequence, overload
+
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from timemachine.parallel.client import AbstractFileClient
+
+
+class StoredArrays(Sequence[NDArray]):
+    """Sequence of numpy arrays using O(1) memory, backed by disk storage.
+
+    Data is stored in a temporary directory that is cleaned when the `StoredArrays` object is finalized.
+
+    Examples
+    --------
+    >>> sa = StoredArrays()
+    >>> sa.extend([np.array([1, 2, 3]), np.array([4, 5, 6])])
+    >>> len(sa)
+    2
+    >>> len(list(sa._chunks()))
+    1
+    >>> sa.extend([np.array([7, 8, 9])])
+    >>> len(sa)
+    3
+    >>> len(list(sa._chunks()))
+    2
+    >>> list(sa)
+    [array([1, 2, 3]), array([4, 5, 6]), array([7, 8, 9])]
+    """
+
+    def __init__(self):
+        self._chunk_sizes: List[int] = []
+        self._dir = Path(tempfile.mkdtemp())
```

Comment:
Should use `tempfile.TemporaryDirectory`? That way we don't need the `weakref.finalize`

Reply:
Good call, that's cleaner (was thinking `TemporaryDirectory` needed to be used as a context manager, but that's not the case). Updated in https://github.com/proteneer/timemachine/pull/950/commits/40b6b1053a19fea968d435b8eb5cefd8d4df8b63